packages/salesforce-adapter/src/filters/custom_objects.ts

```diff
@@ -450,29 +450,20 @@ const createFromInstance = (instance: InstanceElement,
   return [object, ...nestedMetadataInstances]
 }
 
-const fetchSObjects = async (client: SalesforceClient):
-  Promise<Record<string, DescribeSObjectResult[]>> => {
-  const getSobjectDescriptions = async (): Promise<DescribeSObjectResult[]> => {
-    const sobjectsList = await client.listSObjects()
-    const sobjectNames = sobjectsList.map(sobj => sobj.name)
-    return client.describeSObjects(sobjectNames)
-  }
-
-  const getCustomObjectNames = async (): Promise<Set<string>> => {
-    const { result: customObjects } = await client.listMetadataObjects(
-      { type: CUSTOM_OBJECT },
-      // All errors are considered to be unhandled errors. If an error occur, throws an exception
-      () => true
-    )
-    return new Set(customObjects.map(o => o.fullName))
-  }
+const fetchSObjects = async (
+  client: SalesforceClient,
+  customObjectInstances: Record<string, Element>
+): Promise<Record<string, DescribeSObjectResult[]>> => {
+  const sobjectsList = await client.listSObjects()
+
+  const sobjectNames = _(sobjectsList)
+    .map(sobj => sobj.name)
+    .filter(name => name in customObjectInstances)
+    .value()
```

Comment:
nit: no need to use `_`
```suggestion
  const sobjectNames = sobjectsList
    .map(sobj => sobj.name)
    .filter(name => name in customObjectInstances)
```